File: src/client/java/fr/hardel/asset_editor/client/javafx/lib/utils/IconUtils.java
package fr.hardel.asset_editor.client.javafx.lib.utils;

import net.minecraft.resources.Identifier;

public final class IconUtils {

    public static boolean isSvgIcon(Identifier icon) {
        if (icon == null) return false;
        String value = icon.toString();
        return value.startsWith("asset_editor:icons/") || value.endsWith(".svg");
    }

    private IconUtils() {
    }
}

File: src/client/kotlin/fr/hardel/asset_editor/client/compose/lib/utils/IconUtils.kt
package fr.hardel.asset_editor.client.compose.lib.utils

import net.minecraft.resources.Identifier

object IconUtils {

    fun isSvgIcon(icon: Identifier?): Boolean {
        if (icon == null) return false
        val value = icon.toString()
        return value.startsWith("asset_editor:icons/") || value.endsWith(".svg")
    }
}
